gslib/commands/iam.py

```diff
@@ -363,8 +373,106 @@ def get_gcloud_storage_args(self):
       # Gsutil takes policy file then URLs, and gcloud storage does the reverse.
       self.args = url_strings + self.args[:1]
 
+    elif self.sub_command == 'ch':
+      return []  # Translation occurs in self.run_gcloud_storage.
+
     return super().get_gcloud_storage_args(gcloud_storage_map)
 
+  def _RaiseIfInvalidUrl(self, url):
+    if url.IsFileUrl():
+      error_msg = 'Invalid Cloud URL "%s".' % url.object_name
+      if set(url.object_name).issubset(set('-Rrf')):
+        error_msg += (
+            ' This resource handle looks like a flag, which must appear'
+            ' before all bindings. See "gsutil help iam ch" for more details.')
+      raise CommandException(error_msg)
+
+  def _run_gcloud_subprocess(self, args, stdin=None):
+    subprocess_envs = os.environ.copy()
```

Comment:
Swapped the name.

Reply:
I meant the logic

Reply:
Really only two lines are, the ones managing environment variables. The subprocess.run call here has to look quite different, and instead of returning just the return code we need to return the entire process. I could move the lines setting up the subprocess envs to a helper function if you think that'd be better.

Reply:
A DRY helper function in `shim_util.py` would be good

Reply:
Done!